src/views/List.jsx

```diff
@@ -1,18 +1,41 @@
+import { useState } from 'react';
+
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [itemSearch, setItemSearch] = useState('');
+
+	const handleSearchInput = (e) => {
+		setItemSearch(e.target.value.toLowerCase());
+	};
+
+	const handleClear = () => {
+		setItemSearch('');
+	};
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
+			<form>
+				<label htmlFor="itemSearch">Search your shopping list:</label>
+				<input
+					type="text"
+					id="itemSearch"
+					name="itemSearch"
+					value={itemSearch}
+					onChange={handleSearchInput}
+				/>
+				{itemSearch ? (
+					<button onClick={handleClear} aria-label="clear search field">
+						x
+					</button>
+				) : null}
+			</form>
 			<ul>
-				{data.map((data, i) => (
-					<ListItem 
-					key={i} 
-					name={data.name} 
-					/>
-				))}
+				{data
+					.filter((item) => item.name.toLowerCase().includes(itemSearch))
```

Comment:
Do we also need to consider spaces as part of the search criteria?

Reply:
What do you mean by consider?
Right now, both the item name & the search query can have spaces in them

Reply:
oh I see!
hmm. I wonder if this is better solved in the add item component? If we control the input then we could be more standardized with the search.
We could use .trim() but I think that only works for extra spaces at the start and end of the string..

Do you have an idea to solve in mind?

Reply:
Coming in hot with a great solve!
Thanks- looks good to me!

Reply:
OK then, after we've resolved everything from this pull request I will push the changes.

(Aside: Am I correct about the work flow? Would it make more sense for me the push this change right now?)

Reply:
I think you're good to push now, it will just update the code in the PR. We can push again when we resolve the other thing :)